version: "3.3"

services:
  traefik:
    image: "traefik"
    container_name: "traefik"
    command:
      #- "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.websecure.address=:443"
      - "--certificatesresolvers.myresolver.acme.tlschallenge=true"
      - "--certificatesresolvers.myresolver.acme.email=vincent@misson.cloud"
      - "--certificatesresolvers.myresolver.acme.storage=/letsencrypt/acme.json"
    ports:
      - "443:443"
      - "8080:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "/srv/docker-volumes/traefik-letsencrypt:/letsencrypt"
    restart: unless-stopped
    networks:
      - docker    

  portainer:
    image: "portainer/portainer-ce"
    container_name: "portainer"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.portainer-secure.rule=Host(`portainer.cloud63.fr`)"
      - "traefik.http.routers.portainer-secure.entrypoints=websecure"
      - "traefik.http.routers.portainer-secure.tls.certresolver=myresolver"
      - "traefik.http.services.portainer.loadbalancer.server.port=9000"
    volumes:
        - "/var/run/docker.sock:/var/run/docker.sock"
        - "/srv/docker-volumes/portainer-data:/data"
    restart: unless-stopped
    networks:
      - docker

  watchtower:
    image: "containrrr/watchtower"
    container_name: "watchtower"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
    restart: unless-stopped
    networks:
      - docker

networks:
  docker:
    external: true      
